comic/api/mutations.py

```diff
@@ -0,0 +1,96 @@
+import graphene
+
+from comic.api.types import ComicBookInput, ComicBookType, DirectoryInput, DirectoryType
+from comic.models import ComicBook, Directory
+
+
+class CreateDirectory(graphene.Mutation):
+    class Arguments:
+        input = DirectoryInput(required=True)
+
+    ok = graphene.Boolean()
+    directory = graphene.Field(DirectoryType)
+
+    @staticmethod
+    def mutate(root, info, input=None):
+        ok = True
+        directory = Directory(name=input.name)
+        directory.save()
+
+        return CreateDirectory(ok=ok, directory=directory)
+
+
+class UpdateDirectory(graphene.Mutation):
+    class Arguments:
+        id = graphene.Int(required=True)
+        input = DirectoryInput(required=True)
+
+    ok = graphene.Boolean()
+    directory = graphene.Field(DirectoryType)
+
+    @staticmethod
+    def mutate(root, info, id, input=None):
+        ok = False
+        directory = Directory.objects.get(pk=id)
+        if directory:
+            ok = True
+            directory.name = input.name
+            directory.save()
+            return UpdateDirectory(ok=ok, directory=directory)
+        return UpdateDirectory(ok=ok, directory=None)
+
+
+class CreateComicBook(graphene.Mutation):
+    class Arguments:
+        input = ComicBookInput(required=True)
+
+    ok = graphene.Boolean()
+    comic_book = graphene.Field(ComicBookType)
+
+    @staticmethod
+    def mutate(root, info, input=None):
+        ok = True
+        directory = None
+
+        if input.directory:
+            directory = Directory.objects.get(pk=input.directory[0].id)
+
+        comic_book = ComicBook(file_name=input.file_name, directory=directory)
```

Comment:
This should be done by process_comic_book static method. This makes sure all the pages are loaded correctly

Reply:
the way the current system works. you may be better to upload the file first and then trigger the process comic book. it's either that or there will be a need to refactor the way comics are added and processed.